src/Symfony/Component/Messenger/Stamp/LockStamp.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger\Stamp;
+
+use Symfony\Component\Lock\Key;
+
+final class LockStamp implements StampInterface
+{
+    private Key $key;
+    private ?float $ttl;
+    private bool $shouldBeReleasedBeforeHandlerCall;
+
+    public function __construct(
+        string $key,
+        ?float $ttl = 300.0,
```

Comment:
this couples with an unknown handling time

i think it requires #53508

Reply:
> this couples with an unknown handling time
> 
> i think it requires #53508

#53508 might provide an improvement, but is not a blocker.
As a proof, it's been month I'm using this LockStamp without issues.

- If your handling time is short < 300s, there is no issue
- If your handling time is longer, you can still provide a bigger ttl
- If your handling time is longer, you can also consider that if it's been 300s the process started, and it's okay to add a new one in the queue. (If it's the issue you think it can happen ; I'm not even sure I encountered it so far)

At least the LockStamp provide a way to handle race-condition and avoid duplicate in a duration-given (the ttl).

We could consider a merge as-if and another PR after #53508 to improve the feature in a way you have in mind (?)